packages/tupaia-web/src/features/EnlargedDashboardItem/EnlargedDashboardItem.tsx

```diff
@@ -74,29 +77,33 @@ export const EnlargedDashboardItem = ({ entityName }: { entityName?: Entity['nam
     setUrlSearchParams(urlSearchParams);
   };
 
-  const { isExportMode } = exportConfig;
   const isDataDownload =
     (currentDashboardItem?.config as unknown as ViewConfig)?.viewType === 'dataDownload';
 
   const getHasBigData = () => {
     if (isDataDownload || !reportData) return false;
     else if (type === 'matrix') return true;
     const { data } = reportData as BaseReport;
-    return data && data.length > 20;
+    return data ? data?.length > 20 : false;
```

Comment:
Shout out if I’m missing some small-but-important detail, but I *think* this is equivalent and a little easier to parse:

```suggestion
    return data?.length > 20;
```

- If `data` is truthy, then the `?.` is redundant; and
- if its falsy, then `undefined.length > 20` safely evaluates to `false`.

Reply:
There is a weird case where data can sometimes not be an array 😬